conda_build/jinja_context.py

```diff
@@ -29,7 +29,13 @@ def setup(**kw):
         #Patch setuptools
         setuptools_setup = setuptools.setup
         setuptools.setup = setup
-        exec(open(setup_file).read())
+        ns = {
+            '__name__': os.path.splitext(os.path.basename(setup_file))[0],
+            '__doc__': None,
+            '__file__': setup_file,
+        }
+        code = compile(open(setup_file).read(), setup_file, 'exec')
```

Comment:
I think you should pass `dont_inherit=1` here.  Otherwise, this might choke if `setup.py` is written for python 2 and contains a `print` statement.